src/actor/typst.rs

```diff
@@ -176,19 +177,106 @@ impl TypstClient {
 
     async fn process_mail(&mut self, mail: TypstActorRequest) {
         match mail {
-            TypstActorRequest::DocToSrcJumpResolve(id) => {
-                debug!("TypstActor: processing doc2src: {:?}", id);
+            TypstActorRequest::DocToSrcJumpResolve(span_range) => {
+                debug!("TypstActor: processing doc2src: {:?}", span_range);
 
-                let res = self
+                let st_res = self
+                    .inner()
+                    .resolve_span_and_offset(span_range.0.span, Some(span_range.0.offset))
+                    .await
+                    .map_err(|err| {
+                        error!("TypstActor: failed to resolve doc to src jump: {:#}", err);
+                    })
+                    .ok()
+                    .flatten();
+
+                let ed_res = self
                     .inner()
-                    .resolve_span(id)
+                    .resolve_span_and_offset(span_range.1.span, Some(span_range.1.offset))
                     .await
                     .map_err(|err| {
                         error!("TypstActor: failed to resolve doc to src jump: {:#}", err);
                     })
                     .ok()
                     .flatten();
 
+                let loc_leq =
+                    |x: (usize, usize), y: (usize, usize)| x.0 < y.0 || (x.0 == y.0 && x.1 <= y.1);
```

Comment:
https://doc.rust-lang.org/std/primitive.tuple.html#impl-PartialOrd-for-(T,) maybe just use <= ?